tests/cucumber/steps/steps.js

```diff
@@ -4476,6 +4551,384 @@ module.exports = function getSteps(options) {
+
+  When(
+    'I add a method call with the signing account, the current application, suggested params, on complete {string}, current transaction signer, current method arguments.',
+    function (onComplete) {
+      addMethodCallToComposer.call(this, this.signingAccount.addr, onComplete);
+    }
+  );
+
+  When(
+    'I add the current transaction with signer to the composer.',
+    function () {
+      this.composer.addTransaction(this.transactionWithSigner);
+    }
+  );
+
+  When(
+    'I build the transaction group with the composer. If there is an error it is {string}.',
+    function (errorMsg) {
+      assert.strictEqual(errorMsg, ''); // error checking not yet implemented
+      this.composerBuiltGroup = this.composer.buildGroup();
+    }
+  );
+
+  Then('I clone the composer.', function () {
+    this.composer = this.composer.clone();
+  });
+
+  Then(
+    'The composer should have a status of {string}.',
+    function (expectedStatus) {
+      function statusStringToEnum(inString) {
+        switch (inString) {
+          case 'BUILDING':
+            return algosdk.AtomicTransactionComposerStatus.BUILDING;
+          case 'BUILT':
+            return algosdk.AtomicTransactionComposerStatus.BUILT;
+          case 'SIGNED':
+            return algosdk.AtomicTransactionComposerStatus.SIGNED;
+          case 'SUBMITTED':
+            return algosdk.AtomicTransactionComposerStatus.SUBMITTED;
+          case 'COMMITTED':
+            return algosdk.AtomicTransactionComposerStatus.COMMITTED;
+          default:
+            throw Error(
+              `did not recognize AtomicTransactionComposer status string ${inString}`
+            );
+        }
+      }
+
+      assert.strictEqual(
+        this.composer.getStatus(),
+        statusStringToEnum(expectedStatus)
+      );
+    }
+  );
+
+  Then('I gather signatures with the composer.', async function () {
+    this.composerSignedTransactions = await this.composer.gatherSignatures();
+  });
+
+  Then(
+    'the base64 encoded signed transactions should equal {string}',
+    function (commaSeparatedB64SignedTxns) {
+      const expectedSignedTxns = commaSeparatedB64SignedTxns
+        .split(',')
+        .map((b64SignedTxn) => Buffer.from(b64SignedTxn, 'base64'));
+
+      const actualSignedTxns = this.composerSignedTransactions.map(
+        (signedTxn) => Buffer.from(signedTxn)
+      );
+      assert.deepStrictEqual([...actualSignedTxns], [...expectedSignedTxns]);
+    }
+  );
+
+  Then(
+    'I execute the current transaction group with the composer.',
+    async function () {
+      this.composerExecuteResponse = await this.composer.execute(
+        this.v2Client,
+        4
+      );
+      assert.ok(this.composerExecuteResponse.confirmedRound > 0);
+    }
+  );
+
+  Then(
+    'The app should have returned {string}.',
+    function (base64ExpectedReturnValue) {
+      const { methodResults } = this.composerExecuteResponse;
+      assert.strictEqual(methodResults.length, 1);
```

Comment:
non issue, just fyi: I think this could be generalized to checking multiple method returns in the future.

Reply:
Yes I saw the Python implementation does that, which makes a lot of sense. In the interest of getting this merged soon I'll leave it alone for now, but it would be nice to have a general version going forward.